### ISSUE TITLE
[Potential?] memory leak

### ISSUE THREAD

**Issue by UncleGrandpa925** [nomiclabs/hardhat-vscode-feedback#21]
Just putting it here in case anybody encountered it. I have been encountering abnormally high memory usage from VSCode since I update to v.0.0.22. I have never had the issue before, and it only started appearing since the update. I didn't install any new programs to my laptop in the last week or so. 

The issue is on macOS 12.2.1, Macbook 16inch M1 Pro

![memory leak](https://user-images.githubusercontent.com/20136490/154257386-1ec02c3a-b6e8-4ef9-b571-d575300fe455.png)

Most of the time this happens, my laptop will freeze & VSCode is automatically terminated. The issue has happened quite randomly, so I haven't figured out a way to reproduce it. Would really appreciate if the team knows how I can check logs of VSCode to find the source of error.

Thanks a lot!

**Comment by UncleGrandpa925** [comment]
Update: it's happening again after about 1 hour of using VSCode to work on solidity project. Would appreciate a guide to diagnose the issue! 
<img width="281" alt="image" src="https://user-images.githubusercontent.com/20136490/154306856-dc71d914-8cb6-4560-99f9-d9f149ad5de1.png">

**Comment by UncleGrandpa925** [comment]
Update 2: It happened again in about 1 hour (3 times in 3 hours). I have disabled the extension & switched to using Solidity by Juan Blanco and haven't faced the issue ever since

**Comment by kanej** [comment]
Hey @UncleGrandpa925 thanks for the report. Can you point me at the repo you were working on if it is open source?

**Comment by UncleGrandpa925** [comment]
I really want to but unfortunately it's still closed-source at the moment

**Comment by kanej** [comment]
No problem. Does the project pull in any large collection of sol files, openzeppelin for instance?

**Comment by UncleGrandpa925** [comment]
It doesn't contain a lot of solidity files, but yeah it does import the OZ's library

**Comment by kanej** [comment]
Thanks, and I assume you haven't encountered this before last wed (the last release)?

**Comment by UncleGrandpa925** [comment]
Yeah before upgrading I have never faced the issue (I had been using the v0.0.21 since the very beginning and it had zero issues as well)

**Comment by UncleGrandpa925** [comment]
@kanej Oh wow this was fixed. I was waiting for it & haven't seen any replies since then. Thanks a lot!

**Comment by kanej** [comment]
> @kanej Oh wow this was fixed. I was waiting for it & haven't seen any replies since then. Thanks a lot!

@UncleGrandpa925 sorry, that was me. I fixed the memory leak as a priority but forgot to update this thread.

We have released the extension as a public beta if you want to switch: https://marketplace.visualstudio.com/items?itemName=NomicFoundation.hardhat-solidity

And thanks for all the help. It is appreciated.
